File: OneQuestionPerDay/299/bulls_and_cows.py
class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        secret_counter = [0] * 10
        guess_counter = [0] * 10
        bulls = cows = 0
        for i in range(len(secret)):
            if secret[i] == guess[i]:
                bulls += 1
            else:
                secret_counter[ord(secret[i]) - ord('0')] += 1
                guess_counter[ord(guess[i]) - ord('0')] += 1
        for i in range(len(secret_counter)):
            cows += min(secret_counter[i], guess_counter[i])

        return f"{bulls}A{cows}B"

File: OneQuestionPerDay/299/bulls_and_cows.go
package main

import "strconv"

func getHint(secret string, guess string) string {
	secretCount := make([]int, 10)
	guessCount := make([]int, 10)
	bulls := 0
	cows := 0

	for i := 0; i < len(secret); i++ {
		if secret[i] == guess[i] {
			bulls++
		} else {
			secretCount[secret[i]-'0']++
			guessCount[guess[i]-'0']++
		}
	}
	for i, v := range secretCount {
			cows += min(v, guessCount[i])
	}
	return strconv.Itoa(bulls) + "A" + strconv.Itoa(cows) + "B"
}

func min(x, y int) int {
	if x < y {
		return x
	}
	return y
}
